src/components/dataDisplays/DataDisplay.jsx

```diff
@@ -117,6 +125,8 @@ export default function DataDisplay({
     visibleColumnNames.includes(column.name),
   );
 
+  const noResults = showNoResultsBao && data && data.length === 0;
```

Comment:
Is there any possibility for data to be falsey here? If so, maybe something like (howNoResultsBao && data && data.length === 0) || !data
 ?

Reply:
yes, but the `&& data` is the part where that is accounted for. cool thing about javascript is when you have multiple `&&` operators strung together, it will short circuit and return false for the whole expression without evaluating the rest of the line. so if `data=undefined` here you actually never get the `undefined.length does not exist` error